packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -51,13 +91,26 @@ export default function UnsavedInnerBlocks( {
 		}
 	}, [ blocks ] );
 
-	// If the current inner blocks object is different in any way
-	// from the original inner blocks from the post content then the
-	// user has made changes to the inner blocks. At this point the inner
-	// blocks can be considered "dirty".
-	// We also make sure the current innerBlocks had a chance to be set.
-	const innerBlocksAreDirty =
-		!! originalBlocks.current && blocks !== originalBlocks.current;
+	// If the current inner blocks are different from the original inner blocks
+	// from the post content then the user has made changes to the inner blocks.
+	// At this point the inner blocks can be considered "dirty".
+	// Note: referential equality is not sufficient for comparison as the inner blocks
+	// of the page list are controlled and may be updated async due to syncing with
+	// entity records. As a result we need to perform a deep equality check skipping
+	// the page list's inner blocks.
+	const innerBlocksAreDirty = ! isDeepEqual(
+		originalBlocks.current,
+		blocks,
+		( prop, x ) => {
+			// Skip inner blocks of page list during comparison as they
+			// are **always** controlled and may be updated async due to
+			// syncing with enitiy records. Left unchecked this would
+			// inadvertently trigger the dirty state.
+			if ( x?.name && prop === 'innerBlocks' ) {
```

Comment:
```suggestion
			if ( x?.name === 'core/page-list' && prop === 'innerBlocks' ) {
```

Thoughts @draganescu @scruffian ?

Reply:
I'm thinking if you have a social links block and you add a new social link item then it should become dirty. Currently that wouldn't happen.

Reply:
I tested with a social links block, and indeed a new menu is not created in that situation